src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -982,3 +976,18 @@ def get_linux_distro():
             release_info[k.lower()] = v.strip('"')
 
     return release_info.get('name', None), release_info.get('version_id', None)
+
+
+def handle_version_update():
+    try:
+        from azure.cli.core._session import VERSIONS, CLOUD_ENDPOINTS
+        from distutils.version import LooseVersion  # pylint: disable=import-error,no-name-in-module
+        from azure.cli.core import __version__
+        if not VERSIONS['versions']:
+            get_cached_latest_versions()
```

Comment:
the logic is confusing here, what is the object of this function, does it want to refresh the version to latest version or reset the version to empty? If you want to handle some file refresh based on the version, it is better not make it work by some tricky logic

Reply:
The purpose is to detect that azure cli has updated to a new version, then we can invalidate some cached files like version and cloudEndpoints by setting to empty. These files will be refreshed if they are found to be empty in a get operation. No need to refresh them here in advance.

Reply:
should we update the cloudendpoints when CLI upgrades to a new version? It seems cloudendpoints refresh has no relationship with CLI version.

Reply:
cloudendpoints update is not necessarily related to CLI. It could add some endpoints but not used in CLI. But when a new endpoint is also added in CLI, the new version of CLI needs to use the latest endpoints.